apps/rush-lib/src/cli/RushCommandLineParser.ts

```diff
@@ -248,6 +249,9 @@ export class RushCommandLineParser extends CommandLineParser {
 
     this._validateCommandLineConfigCommand(command);
 
+    const overrideAllowWarnings: boolean =
+      this.rushConfiguration && EnvironmentConfiguration.allowWarningsInSuccessfulBuild;
```

Comment:
Why do we need `this.rushConfiguration &&` here?

Reply:
Was working with Emmanuel on this one... this is early enough in the command line parser that it's possible rush.json doesn't exist (for example, we might be preparing to run "rush init" in an empty folder).

Because we don't know what command we are about to run, it's possible that rush configuration is undefined at this point, and if rush configuration is undefined then the EnvironmentConfiguration is not initialized.

Reply:
This makes a assumption that `rushConfiguration !== undefined` implies that `EnvironmentConfiguration` is initialized, which seems brittle.  (Is that guaranteed? Is that the only way it can be initialized?)

Maybe a simpler solution is to ensure that the `EnvironmentConfiguration` is initialized before we access it.

Reply:
Was chatting more about this offline and I agree, and I think what we might do is:

1. Change `EnvironmentConfiguration` so that it initializes itself if it hasn't been initialized yet.
2. Continue to initialize (without reset) at key narrow paths like RushConfiguration, where it makes sense to capture and error on the majority of possible configuration errors.
3. Possibly some other small tweaks to the error behavior of EnvironmentConfiguration.

These changes (especially #3) probably deserve their own dedicated review, so if it's cool with you we can make those changes in a follow-up PR.